.github/workflows/test-native-tooling-packaging.yml

```diff
@@ -0,0 +1,75 @@
+name: Test native tooling packaging
+
+# This workflow tests packaging of Fleet-osquery with the
+# `fleetdm/fleetctl` Docker image. 
+
+on:
+  push:
+    branches:
+      - main
+      - patch-*
+  pull_request:
+    paths:
+      - '**.go'
+      - 'tools/fleetctl-docker/**'
+      - 'tools/wix-docker/**'
+      - 'tools/bomutils-docker/**'
+      - '.github/workflows/test-native-tooling-packaging.yml'
+  workflow_dispatch: # Manual
+
+permissions:
+  contents: read
+
+jobs:
+  test-packaging:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        go-version: ['^1.17.8']
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@b22fbbc2921299758641fab08929b4ac52b32923 # v2
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    - name: Checkout Code
+      uses: actions/checkout@629c2de402a417ea7690ca6ce3f33229e27606a5 # v2
+
+    - name: Install Go Dependencies
+      run: make deps-go
+
+    - name: Build fleetctl
+      run: make xp-fleetctl
+
+    # TODO: pull the image from the registry instead once it's published 
```

Comment:
not sure about this TODO, maybe it's a good idea to build the image since it will include a fresh `fleetctl` (I would probably tweak the `paths` to be more cautious when triggering this)